### PR TITLE
Twg

### PR DESCRIPTION
**Pull request by zxianshu**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by iakov-aws — CHANGES_REQUESTED** [review]

File: static/Cost/300_Optimization_Data_Collection/Code/source/tg.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+s3_client = boto3.client('s3')
+today = date.today()
+year = today.year
+month = today.month
+
+def lambda_handler(event, context):
+    event = {'Records': [{'messageId': 'a520aa0d-9b42-4ee3-8ee9-906feee83c69', 'receiptHandle': 'AQEBmuXPrJ+kjS1/e8eg6nqhkkmSk8SWoaLgv1wgWaPQVkfVjLdJvZzcb8wbh4Yug5WajIy9k7M7LC6rRLtz+Qu/sUTDUX42CZOdSK/kEOFsZKfjoXoqNJHyatOOo7KZQluAl7Ao2tz1ls5+hpZtF2QhjNQFRE9BWDUmMSGHGvVjuNWUP0BDa7/qent4y5FhkSGcGDrtcZX7PbbzherQdhK5McqqgfC6sCHmKA6zwn9CSWxiPpGU2MSAgHxfDjsbjryY9rWR9CJRduhV8MYBCpqjqZa+eZfgMuhcFaK2COTmQOEQSqJkFBnFco+tFYG8meE0x9tQ5EDg67k+8l7WZp57iT15wzh0RpMy4fdlSVw/VSujtVFPfaOrW4phLpC/DTpp+27DfeTQ1XGGelRI9UKUru1/1pXM1rsWn4Zc9CzVBYs=', 'body': '{"account_id": "806331864324", "account_name": "xianshutrainingbilling"}', 'attributes': {'ApproximateReceiveCount': '1', 'SentTimestamp': '1664789470228', 'SenderId': 'AROAYQJSXGSHNS7FPPLME:AWS-Organization-Account-Collector', 'ApproximateFirstReceiveTimestamp': '1664789472228'}, 'messageAttributes': {}, 'md5OfBody': '49159a8328f27c1bb3ccf038332deaed', 'eventSource': 'aws:sqs', 'eventSourceARN': 'arn:aws:sqs:us-east-1:584758801550:TransitGatewayModule-TaskQueue', 'awsRegion': 'us-east-1'}]}
+
+    print(event)
```
[the remark this answers is not in the document]
Comment (awssteph) [reply static/Cost/300_Optimization_Data_Collection/Code/source/tg.py:23]: thank you

File: static/Cost/300_Optimization_Data_Collection/Code/source/tg.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        # Send some context about this error to Lambda Logs
+        logging.warning("%s" % e)
+
+def get_supported_regions():
+    response = []              
+    ec2_c = boto3.client('ec2')
+    response = ec2_c.describe_regions()
+    return response['Regions'] if response['Regions'] else []
+
+lambda_handler(None, None)
```
[the remark this answers is not in the document]
Comment (awssteph) [reply static/Cost/300_Optimization_Data_Collection/Code/source/tg.py:158]: thanks

File: static/Cost/300_Optimization_Data_Collection/Code/source/tg.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                            print (cw_results)
+                            jsondata = json.dumps(cw_results) 
+                            f.write(jsondata)
+                            f.write('\n')
+                                                
+                    s3 = boto3.client("s3", config=Config(s3={"addressing_style": "path"}))
+                    s3.upload_file(
+                        "/tmp/data.json",
+                        bucket,
+                        f"{os.environ['TABLE']}/year={year}/month={month}/tgw-{item['TransitGatewayAttachmentId']}-{region_name}.json")
```
[the remark this answers is not in the document]
Comment (awssteph) [reply static/Cost/300_Optimization_Data_Collection/Code/source/tg.py:74]: will be but not sure which PR is in first as multi payer is not merged

**Comment by zxianshu** [comment]
Thanks @iakov-aws I've deleted that file. It is a test file and we forgot to delete it.

File: static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+  CURTable:
+    Type: String
+    Description: The name of your Cost and Usage Report table in Athena
+    Default: test_usage
+  DestinationBucketARN:
+    Type: String
+    Description: ARN of the S3 Bucket that exists or needs to be created to hold rightsizing information
+  GlueRoleArn:
+    Type: String
+    Default: "cron(30 12 1 * ? *)"
```
Comment (yprikhodko) [inline static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml:27]: seems like a wrong default value for GlueRoleARN param

File: static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+  CFDataName:
+    Type: String
+    Description: The name of what this cf is doing.
+    Default: transitgateway
+  DatabaseName:
+    Type:  String
+    Description: Default CUR database name
+    Default: optimization_data
+    MinLength: 1
+  CURTable:
```
[the remark this answers is not in the document]
Comment (zxianshu) [reply static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml:18]: upgraded to be the same as other modules

File: static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml (hunk 4 above)
Comment (zxianshu) [reply static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml]: updated

Comment (zxianshu) [reply static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml]: @yprikhodko updated

File: static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (zxianshu) [reply static/Cost/300_Optimization_Data_Collection/Code/transit_gateway.yaml:18]: @yprikhodko

**Review by yprikhodko — APPROVED** [review]

File: static/Cost/300_Optimization_Data_Collection/Code/Optimization_Data_Collector.yaml [hunk 6]
```diff
@@ -408,6 +422,17 @@ Resources:
...
+    Type: AWS::CloudFormation::Stack
+    Condition: DeployTransitGatewayModule
+    Properties:
+      Parameters:
+          DestinationBucket: !Ref S3Bucket
+          DestinationBucketARN: !GetAtt S3Bucket.Arn 
+          GlueRoleArn: !GetAtt GlueRole.Arn 
+          MultiAccountRoleName: !Sub "${RolePrefix}${MultiAccountRoleName}"
+          RolePrefix: !Ref RolePrefix
+      TemplateURL: "https://aws-well-architected-labs.s3-us-west-2.amazonaws.com/Cost/Labs/300_Optimization_Data_Collection/transit_gateway_dev.yaml"
```
Comment (awssteph) [inline static/Cost/300_Optimization_Data_Collection/Code/Optimization_Data_Collector.yaml:435]: remove Dev and we wil be ready to go

**Review by awssteph — APPROVED** [review]

**Pull request merged**